tools/ci/ci_demos.py

```
@@ -0,0 +1,196 @@
+def _cli_invoke_next(args_list, prefix="cli"):
+    next_cmd = args_list.func
+    args = args_list
+    func = globals()[prefix + "_" + next_cmd.replace("-", "_")]
+    func_args = [args] if vars(args) else [] + [args_list] if args_list else []
+    return func(*func_args)
+
+
+def _config_build(_src, _build_flags, _build_path):
+    _build_flags = " ".join(_build_flags)
+    _run_cmd(f'cmake -S {_src} -B {_build_path} {_build_flags}  -G "Unix Makefiles"')
+
+
+def _build_target(_target, _c_flags, build_path):
+    print("\n----------------------------------------------------------------")
+    print(f"Building target: {_target}")
+    print("----------------------------------------------------------------")
+    _c_flags = " ".join(_c_flags)
+    cmd = f"make -C {build_path} {_c_flags} {_target}"
+    print(_run_cmd(cmd))
+
+
+def _get_targets(build_path):
+    _targets = _run_cmd(f"make help -C {build_path} | tr -d '. '")
+    _targets = [_t.strip() for _t in _targets.split()]
+    return _targets
+
+
+def _build_targets(_targets, _config, build_path):
+    result = {}
+
+    for _target in _targets:
+        _target_result = result.setdefault(_target, {})
+        try:
+            _c_flags = _config.get("_default", {}).get("c_flags", []) + _config.get(
+                _target, {}
+            ).get("c_flags", [])
+            _build_target(_target, _c_flags, build_path)
+            _target_result["status"] = "PASS"
+        except subprocess.CalledProcessError as err:
+            print(err.stdout)
+            _target_result["status"] = "FAIL"
+    return result
+
+
+def _del_dir(dir_name):
+    try:
+        print(f"Deleting dir: {dir_name}")
+        shutil.rmtree(dir_name)
+    except OSError as err:
+        print("Error: %s - %s." % (err.filename, err.strerror))
+
+
+def _read_config(_path):
+    try:
+        _config = yaml.load(_path.read_text())
+        return _config
+    except:
```

bare except. Might want to use `except Exception:`